openforcefield/utils/toolkits.py

```diff
@@ -3438,6 +3442,360 @@ def assign_partial_charges(
+
+        rdkit_molecule = molecule_copy.to_rdkit()
+
+        # Check for trans-COOH configurations
+        carboxylic_acid_matches = cls._find_smarts_matches(
+            rdkit_molecule, "[#6X3:2](=[#8:1])(-[#8X2H1:3]-[#1:4])"
+        )
+
+        for match in carboxylic_acid_matches:
+
+            dihedral_angle = GetDihedralRad(rdkit_molecule.GetConformer(0), *match)
+
+            if dihedral_angle > np.pi / 2.0:
+                # Discard the 'trans' conformer.
+                return (
+                    True,
+                    "Molecules which contain COOH functional groups in a trans "
+                    "configuration are discarded by the ELF method.",
+                )
+
+        return False, None
+
+    @classmethod
+    def _elf_prune_problematic_conformers(
+        cls, molecule: "Molecule"
+    ) -> List[unit.Quantity]:
+        """A function which attempts to remove conformers which are known to be
+        problematic when computing ELF partial charges.
+
+        Currently this includes conformers which:
+
+        * contain a trans-COOH configuration. These conformers ... TODO add reason.
+
+        Notes
+        -----
+        * Problematic conformers are flagged by the
+          ``RDKitToolkitWrapper._elf_is_problematic_conformer`` function.
+
+        Returns
+        -------
+            The conformers to retain.
+        """
+
+        valid_conformers = []
+
+        for i, conformer in enumerate(molecule.conformers):
+
+            is_problematic, reason = cls._elf_is_problematic_conformer(
+                molecule, conformer
+            )
+
+            if is_problematic:
+                logger.warning(f"Discarding conformer {i}: {reason}")
+            else:
+                valid_conformers.append(conformer)
+
+        return valid_conformers
+
+    @classmethod
+    def _elf_compute_electrostatic_energy(
+        cls, molecule: "Molecule", conformer: unit.Quantity
+    ) -> float:
+        """Computes the 'electrostatic interaction energy' of a particular conformer
+        of a molecule.
+
+        The energy is computed as the sum of ``|q_i * q_j| * r_ij^-1`` over all pairs
+        of atoms (i, j) excluding 1-2 and 1-3 terms, where q_i is the partial charge
+        of atom i and r_ij the Euclidean distance between atoms i and j.
+
+        Notes
+        -----
+        * The partial charges will be taken from the molecule directly.
+
+        Parameters
+        ----------
+        molecule
+            The molecule containing the partial charges.
+        conformer
+            The conformer to compute the energy of. This should be a unit wrapped
+            numpy array with shape=(n_atoms, 3) with units compatible with angstroms.
+
+        Returns
+        -------
+            The electrostatic interaction energy in units of [e^2 / Angstrom].
+        """
+
+        if molecule.partial_charges is None:
+            raise ValueError("The molecule has no partial charges assigned.")
```

Comment:
I think this will always be more of an internal exception (i.e. the user should never see this raised) so I think I'll leave it as is for now, but can maybe revisit this once you do the exception overhaul.